### PR TITLE
dockerized wordcount example

### PR DESCRIPTION
**Pull request by sebastianneubauer**
This add a docker-compose project to the wordcount example. Additionally to the native execution, the very same example can now be executed with docker in several services:
* kafka: a kafka server
* kafka-manager: a gui to see what is happening in kafka (configure the zookeeper: kafka:2181)
* wordcount: the wordcount kafka streams example app
* source_client: a trivial client which writes to kafka
* kafka-debug: a trivial consumer for the kafka output stream of the wordcount example

If you like this, one can easily dockerize the other examples and even the integration tests can be automated using docker-compose

If you have any questions, just ask, this is an early demo, I am happy to incorporate any suggestions...

### CODE REVIEW

**Comment by ah-** [comment]
Cool, thanks! Will give this a go tomorrow.

File: examples/wordcount/docker/source_client/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+ADD . /code
+
+RUN apt-get update
+RUN echo "/usr/local/lib" >> /etc/ld.so.conf
+RUN git clone https://github.com/edenhill/librdkafka.git /tmp/librdkafka
+RUN ls /tmp/ && cd /tmp/librdkafka && ./configure && make && make install && ldconfig
+
+WORKDIR /code/examples/wordcount/
+RUN pip --version
+#RUN pip install -e git+https://github.com/confluentinc/confluent-kafka-python.git#egg=confluent-kafka
```
Comment (ah-) [inline examples/wordcount/docker/source_client/Dockerfile]: Just gave this a go, and got this:
```
$ docker logs docker_wordcount_1
Traceback (most recent call last):
  File "example.py", line 13, in <module>
    from winton_kafka_streams.processor import BaseProcessor, TopologyBuilder
  File "/code/winton_kafka_streams/processor/__init__.py", line 10, in <module>
    from ._stream_thread import StreamThread
  File "/code/winton_kafka_streams/processor/_stream_thread.py", line 10, in <module>
    from confluent_kafka import KafkaError
ModuleNotFoundError: No module named 'confluent_kafka'
```

Should this be commented in?

File: examples/wordcount/docker/source_client/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+RUN pip install -e ../../
+
+CMD ["python", "source_client.py"]
```
Comment (ah-) [inline examples/wordcount/docker/source_client/Dockerfile]: Getting this:

```
$ docker logs docker_source_client_1
python: can't open file 'source_client.py': [Errno 2] No such file or directory
```

I guess this file is just missing, could you add it?

File: examples/wordcount/docker/wordcount/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+ADD . /code
+
+RUN apt-get update
+RUN echo "/usr/local/lib" >> /etc/ld.so.conf
+RUN git clone https://github.com/edenhill/librdkafka.git /tmp/librdkafka
+RUN ls /tmp/ && cd /tmp/librdkafka && ./configure && make && make install && ldconfig
+
+WORKDIR /code/examples/wordcount/
+RUN pip --version
+#RUN pip install -e git+https://github.com/confluentinc/confluent-kafka-python.git#egg=confluent-kafka
```
Comment (ah-) [inline examples/wordcount/docker/wordcount/Dockerfile]: Probably the pip install below should pull this in, do you remember why you have the commented out pip install confluent-kafka-python here?

Getting this:
```
$ docker logs docker_wordcount_1
Traceback (most recent call last):
  File "example.py", line 13, in <module>
    from winton_kafka_streams.processor import BaseProcessor, TopologyBuilder
  File "/code/winton_kafka_streams/processor/__init__.py", line 10, in <module>
    from ._stream_thread import StreamThread
  File "/code/winton_kafka_streams/processor/_stream_thread.py", line 10, in <module>
    from confluent_kafka import KafkaError
ModuleNotFoundError: No module named 'confluent_kafka'
```

Comment (llawall) [inline examples/wordcount/docker/wordcount/Dockerfile]: I think the reason this does not get pulled in automatically for us might be a quirk in our local setup @ah-

File: examples/wordcount/docker/source_client/Dockerfile (hunk 2 above)
Comment (sebastianneubauer) [inline examples/wordcount/docker/source_client/Dockerfile]: Sorry, my fault...added now!

File: examples/wordcount/docker/wordcount/Dockerfile (hunk 3 above)
Comment (sebastianneubauer) [inline examples/wordcount/docker/wordcount/Dockerfile]: `confluent_kafka` is a dependency in your setup.py, so with `RUN pip install -e ../../` it should be pulled in...Can you find out if it doesn't get installed (`RUN pip freeze`) or if there is a different problem? you can build the service with `docker-compose build wordcount`

**Comment by sebastianneubauer** [comment]
Great thanks! If I have time I will look into the other examples too...
What about the tests, are there integration-tests that currently need a running kafka to succeed? That could be addressed by docker-compose too...

**Comment by ah-** [comment]
Yes, that sounds like a good idea. Initially was going to look at https://github.com/dpkp/kafka-python/blob/master/test/fixtures.py and https://github.com/edenhill/trivup and just re-use one of them, but Docker would make it a bit easier to get it to work reliably.

**Comment by llawall** [comment]
@sebastianneubauer : a quick note to say that we are considering requiring a CLA to be signed for contributions so I'll be in touch next week if this goes ahead. 

Looking forward to your docker-compose based test harness - this will be a big help to ensure consistent testing.

**Pull request merged**